### ISSUE TITLE
[BUG] RGBW2 binary_sensor for switch not reporting correct value (FW 1.8.3)

### ISSUE THREAD

**Issue by david-kalbermatten** [StyraHem/ShellyForHASS#408]
## Environment

- ShellyForHASS release with the issue: 0.2.0.beta.3
- Last working ShellyForHASS release (if known): 1.9
- Home Assistant Core release with the issue: 0.114.4
- Operating environment (Home Assistant/Supervised/Docker/venv): Docker

## Describe the bug

After updating to the new beta build of ShellyForHass and updating one of my Shelly RGBW2 is now showing an odd behavior where it cycles the state of the switch sensor On/Off/On/Off. I can confirm with a second RGBW2 that this issue only shows up on the new firmware.

## Steps to Reproduce

1. Update Shelly RGBW2 to newest firmware (1.8.3 at the time of writing)
2. Observe `binary_sensor.shelly_shrgbw2_xxxxxxxx_switch`

## Expected behavior

It should represent the actual signal of the switch hooked up to it :D

**Comment by ttajmajer** [comment]
I observe exactly the same problem. Shelly RGBW2 configured as 4 white channels. 
Another RGBW2 with firmware v1.6.0 in the same network is working fine.

0.2.0.beta.2
RGBW2 firmware: 20200818-120632/v1.8.2@36539b0b
Home Assistant 0.114.3

**Comment by hakana** [comment]
Try beta 4

**Comment by david-kalbermatten** [comment]
Seems to work as expected now. I'll test it more extensively in the evening

**Comment by hakana** [comment]
👍

**Comment by david-kalbermatten** [comment]
Alright, tested it. Everything seems to work as it should, this includes correct and timely updates of the binary_sensor value of the switch and both event types fire correctly and immediately. Although (just like I mentioned in issue #409) the click_types for `shellyforhass.click` are limited to `single` and `long`

**Comment by ttajmajer** [comment]
I've tested the beta 4. Now RGBW2 with firmware 1.8+ works fine, but the second RGBW2 with firmware 1.6 is reporting switch toggle on/off all the time. 
I'll update to the newest firmware, however maybe the old one should be supported in the code as well?

**Comment by david-kalbermatten** [comment]
I haven't updated one of my Shelly RGBW2s which means it's still on 1.7.3 and it's not showing the on/off behavior. Please make sure you actually have beta4 installed and restarted home assistant before testing @ttajmajer 

TLDR: For me both firmware versions `1.7.3` and `1.8.3` work just fine with beta4

**Comment by ttajmajer** [comment]
I've updated both to 1.8.3 and I don't observe on/off behavior anymore. It's hard to make a firmwared downgrade on RGBW2s, so it will be hard to test if firmware 1.6 has this problem. 
For me the issue is solved.
